rfc/reproducible-builds.md

```diff
@@ -0,0 +1,66 @@
+# Reproducible Builds
+To ensure the security of constellation's supply chain, we need to make our software builds reproducible.
+This is the only way to ensure a verifiable path from source code to binary.
+Every step of the build process must be deterministic.
+
+## Definition of Goals
+1. All our OCIs executed in the cluster must be reproducible bit by bit .
+1. For that, every compiled executable has to be deterministically compiled.
+1. If we change parts of our codebase (i.e. add several other programming languages) we should not have to make major changes to the image build system.
+1. Moving from `Docker` to a reproducible build system should have a minimum overhead. Ideally, we can reuse our existing `Dockerfile`s.
```

Comment:
We could state why we are moving away from `Docker` or more precisely how `Docker` breaks reproducibility.